.gitlab-ci.yml

```diff
@@ -91,6 +91,18 @@ cargo-audit:
   allow_failure:                   true
 
 
+check-warnings:
+  stage:                           test
+  <<:                              *docker-env
+  except:
+    - /^[0-9]+$/
+  script:
+    - ./scripts/build.sh --locked
+    - RUSTFLAGS="-D warnings" time cargo check --all --release --verbose
+    - sccache -s
+  allow_failure:                   true
```

Comment:
Should failures be allowed here?

Reply:
@DemiMarie-parity 
I'm going to rewrite this completely, will take the log from, say `cargo test`, and parse it for warnings later.
Answering to your question, yes, in this case it should be allowed, at least for the fist time. We have lots of warnings there and `RUSTFLAGS="-D warnings"` turns any warning into an error and stops. Which is not desired for the most PRs.